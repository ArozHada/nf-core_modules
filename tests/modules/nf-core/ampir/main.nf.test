nextflow_process {

    name "Test Process AMPIR"
    script "modules/nf-core/ampir/main.nf"
    process "AMPIR"
    config "tests/modules/nf-core/ampir/nextflow.config"
    tag "ampir"

    test("Candidatus Portiera Aleyrodidarum Proteome Fasta") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test', single_end:false ], // meta map
                               file(params.test_data['candidatus_portiera_aleyrodidarum']['genome']['proteome_fasta'], checkIfExists: true),
                           ]
                input[1] = "precursor"
                input[2] = []
                input[3] = "0.7"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }
    }

}
