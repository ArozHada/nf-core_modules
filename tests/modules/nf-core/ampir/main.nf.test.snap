{
    "Candidatus Portiera Aleyrodidarum Proteome Fasta": {
        "content": [
            [
                "test.faa:md5,0435609144022c55ac196db053f0df89",
                "test.tsv:md5,83e6dfa351935981bf54d9ce65b4a20c",
                "versions.yml:md5,f8d5026ccdd8f72c7ac1b5e4670aab49"
            ]
        ],
        "timestamp": "2023-04-06T00:38:27+0000"
    }
}